frontend/src/app/main/data/workspace/groups.cljs

```diff
@@ -150,6 +153,43 @@
         (pcb/change-parent parent-id children idx-in-parent)
         (pcb/remove-objects [(:id frame)]))))
 
+
+(defn- clone-component-shapes-changes
+  [changes shape objects]
+  (let [[new-shape new-shapes _]
+        (ctst/clone-object shape
+                           (:parent-id shape)
+                           objects)
+
+        new-shape-id (:id new-shape)
+        new-shapes (->> new-shapes
+                        (filter #(not= (:id %) new-shape-id))
+                        (map #(if (= new-shape-id (:parent-id %))
+                                (assoc % :parent-id (:parent-id shape))
+                                %)))]
```

Comment:
I think you may do this in `update-new-object` callback of  `clone-object`: this callback is for doing this kind of things. And so, you avoid a map. You will need only the filter.

Additionally, I think it's cleaner if you replace the `if` by a `cond->`.

Reply:
Changed